### PR TITLE
feat: Add detection for recovered 'expected to fail' scrapers

### LINKED ISSUE #146: NetivHased scraper points to dead IP; retailer has moved to https://app.netiv-hesed.com/
## Description

Two chains are hardcoded in `ScraperStability` to always report "expected to fail"
regardless of actual site status — but both appear to be live and serving real,
current data as of today (2026-08-29).

### NetivHased

`NetivHased` scrapes a hardcoded IP:

```python
class NetivHased(WebBase):
    def __init__(self, file_output=None, status_database=None):
        super().__init__(
            chain=DumpFolderNames.NETIV_HASED,
            chain_id="7290058160839",
            url="http://141.226.203.152/",
            ...
        )
```

and is marked permanently failing in `ScraperStability.NetivHased`:

```python
class NetivHased(FullyStable):
    """Netiv Hased site is down (HTTP 500 on http://141.226.203.152/).
    Evidence: upstream returns HTTP 500 for store/price/promo scrapes as of
    2026-07-24 ... site is now unavailable on weekdays too.
    """
    @classmethod
    def failire_valid(cls, **_):
        return True
```

But gov.il's official retailer-links page
(https://www.gov.il/he/pages/cpfta_prices_regulations) currently lists NetivHased's
portal as **`https://app.netiv-hesed.com/`** — a different domain, not the hardcoded
IP. I checked this URL directly today: it's live, serving files for previous days
(none for today, which is a Saturday — consistent with a kosher-observant retailer
not publishing on Shabbat, not a dead site). This strongly suggests the retailer
migrated hosts, the old IP started 500ing, and the scraper was never updated to
follow — rather than the site being genuinely down.

### CityMarketKiryatGat

`ScraperStability`'s `CityMarketKiratGat.failire_valid` also unconditionally forces
failure:

```python
@classmethod
def failire_valid(cls, when_date=None, files_types=None, utilize_date_param=True, **_):
    return (
        super(cls, CityMarketKiratGat).failire_valid(...)
        or True
    )  # there is an active issue with the site
```

I checked its gov.il-listed portal (`https://citymarketkiryatgat.binaprojects.com/Main.aspx`)
directly today: it's live and actively publishing files with today's date and hourly
timestamps, e.g. `StoresFull7290058266241-000-202608291000.gz` (10:01, 29/08/2026).
No sign of an active issue.

## Suggested fix

- Update `NetivHased`'s `url` to `https://app.netiv-hesed.com/` (adjusting for
  whatever file-listing structure the new site uses), and re-evaluate whether
  `ScraperStability.NetivHased.failire_valid` still needs to unconditionally return
  `True` once the new URL is confirmed working.
- Re-evaluate `CityMarketKiratGat.failire_valid`'s unconditional `or True` — the site
  appears to be functioning normally today; happy to help narrow down when the
  "active issue" comment was added if that helps confirm whether it's since been
  resolved upstream.

Both overrides may have been accurate when written, but appear stale now. Since both
currently cause these chains to be silently skipped when scraping all chains (and to
hard-crash with `ValueError` if explicitly named in `enabled_scrapers`, per a related
inconsistency — see if that's worth its own issue), fixing or re-verifying these
would restore 2 of the 32 legally-required retailers to working coverage.

## Environment

- Observed live via browser, 2026-08-29:
  - https://app.netiv-hesed.com/ (NetivHased's current gov.il-listed URL)
  - https://citymarketkiryatgat.binaprojects.com/Main.aspx (CityMarketKiryatGat's gov.il-listed URL)
- il_supermarket_scarper: 1.0.8

### PR DESCRIPTION
**Pull request by erlichsefi**
## Summary

This PR adds comprehensive detection for scraper stability issues and creates a single source of truth for deprecated scrapers via `DEPRECATED_SCRAPERS` constant.

### Changes

**1. `DEPRECATED_SCRAPERS` Constant** (`scrappers_factory.py`)
- Single source of truth for all deprecated scrapers
- All deprecated scrapers in enum but excluded via `all_listed_scrappers()`
- Added `get_deprecated_scrapers()` and `is_deprecated()` methods

**2. `DumpFolderNames.active_folder_names()`** (`folders_name.py`)
- Now uses `DEPRECATED_SCRAPERS` from factory (lazy import to avoid circular dependency)
- Removed duplicate hardcoded set

**3. `ScraperKind` Classification** (`scraper_stability.py`)
- `EDGE_CASE`: Normal scrapers with known edge cases
- `ALWAYS_FAILING`: On gov.il but currently broken
- `DEPRECATED`: Not on gov.il anymore

### Deprecated Scrapers (Single Source of Truth)

| Scraper | Reason |
|---------|--------|
| `VICTORY` | Replaced by `VICTORY_NEW_SOURCE` |
| `HET_COHEN` | Replaced by `HET_COHEN_NEW_SOURCE` |
| `MAHSANI_ASHUK` | Replaced by `MAHSANI_ASHUK_NEW_SOURCE` |
| `COFIX` | Folded into Rami Levy |
| `MEGA` | Merged with other chains |
| `QUIK` | Site down, removed from gov.il |
| `CITY_MARKET_GIVATAYIM` | Closed |
| `CITY_MARKET_KIRYATONO` | Closed |

Fixes #146

<div><a href="https://cursor.com/agents/bc-dbc8abb4-eefe-4c01-98fa-6e666cee53de?cursor_ref=pr_footer&cursor_cta=open_in_web"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-dbc8abb4-eefe-4c01-98fa-6e666cee53de&cursor_ref=pr_footer&cursor_cta=open_in_cursor"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

File: il_supermarket_scarper/scraper_stability.py [hunk 1]
```diff
@@ -12,9 +14,30 @@
 from il_supermarket_scarper.utils.logger import Logger
 
 
+def _load_gov_il_links():
```
Comment (erlichsefi) [inline il_supermarket_scarper/scraper_stability.py]: this should be parsed out from the html cpfta_prices_regulations.

ignore scripts.

File: il_supermarket_scarper/scraper_stability.py [hunk 2]
```diff
@@ -67,13 +90,18 @@ def failire_valid(cls, **_):
...
-    site is now unavailable on weekdays too.
+    Evidence: upstream returns HTTP 500 for http://141.226.203.152/ as of
+    2026-07-24 (CI NetivHasefTestCase all failing). However, gov.il lists
+    the retailer at https://app.netiv-hesed.com/ which is live and serving files.
+
+    Status: UNSTABLE (not deprecated) - retailer still on gov.il, URL needs update.
     """
 
+    is_deprecated = False
+    expected_gov_il_url = "https://app.netiv-hesed.com/"
```
Comment (erlichsefi) [inline il_supermarket_scarper/scraper_stability.py]: should be taken from the scraper class.

File: il_supermarket_scarper/scraper_stability.py [hunk 3]
```diff
@@ -12,9 +14,30 @@
...
+
 class FullyStable:
     """fully stable is stablity"""
 
+    # If True, the retailer is no longer on gov.il and scraper is expected to not work
+    is_deprecated = False
+
+    # Expected URL from gov.il (if different from scraper's hardcoded URL)
+    # Set this when gov.il lists a different URL than what the scraper uses
+    expected_gov_il_url = None
```
Comment (erlichsefi) [inline il_supermarket_scarper/scraper_stability.py]: should be taken from the scraper class.

File: il_supermarket_scarper/scraper_stability.py [hunk 4]
```diff
@@ -154,14 +191,11 @@ def failire_valid(
...
-                files_types=files_types,
-                utilize_date_param=utilize_date_param,
-            )
-            or True
-        )  # there is an active issue with the site
+        return super(cls, CityMarketKiratGat).failire_valid(
+            when_date=when_date,
+            files_types=files_types,
+            utilize_date_param=utilize_date_param,
+        ) or cls.searching_for_update_promo_full(files_types=files_types)
```
Comment (erlichsefi) [inline il_supermarket_scarper/scraper_stability.py]: why did you added this?

File: il_supermarket_scarper/scraper_stability.py (hunk 1 above)
Comment (erlichsefi) [reply il_supermarket_scarper/scraper_stability.py]: Done. `validate_against_gov_il` now parses retailer hrefs from the cached `cpfta_prices_regulations` HTML (`get_cpfta_retailer_links`). It no longer reads `scripts/gov_il_links.json`.

File: il_supermarket_scarper/scraper_stability.py (hunk 2 above)
Comment (erlichsefi) [reply il_supermarket_scarper/scraper_stability.py]: Done. Removed `expected_gov_il_url`. Drift is now `scraper.url` (from the scraper class) vs hosts parsed from the cpfta HTML.

File: il_supermarket_scarper/scraper_stability.py (hunk 3 above)
Comment (erlichsefi) [reply il_supermarket_scarper/scraper_stability.py]: Done. Dropped the class attribute. The URL comes from the scraper class, not from stability config.

File: il_supermarket_scarper/scraper_stability.py (hunk 4 above)
Comment (erlichsefi) [reply il_supermarket_scarper/scraper_stability.py]: Removed it. That `or searching_for_update_promo_full` was a guess after dropping `or True`. Restored the original `or True` so we don't invent a narrower fail condition without evidence.

File: il_supermarket_scarper/tests/test_scrappers_factory.py [hunk 5]
```diff
@@ -17,12 +17,130 @@ def test_stable_scraper():
...
+
+    assert actual_not_active == expected_not_active, (
+        f"Not-active scrapers mismatch. "
+        f"Expected: {expected_not_active}, "
+        f"Actual: {actual_not_active}. "
+        f"Difference: {actual_not_active.symmetric_difference(expected_not_active)}"
+    )
+
+
+def test_permanently_failing_scrapers_documented():
```
Comment (erlichsefi) [inline il_supermarket_scarper/tests/test_scrappers_factory.py]: this should now be available by checking which scrapers is deprecated (those are         "QUIK",
        "VICTORY")



"CITY_MARKET_KIRYATGAT",
        "NETIV_HASED",

links (in the calss) should be failed to found in the gov.il site and thus a test should fail.

**Comment by erlichsefi** [comment]
Let's revisit the entire PR strategy here. Two scrappers were identified where the source in Gov il deviated but they passed since we allowed them to fail all together. 

The goal of this change is to introduce three configurations of a scraper. 
One is a functioning scrapper that is listed in the gov website and there is edge cases where it not provided data.

One is scrapper that is already removed from gov il website and we keeping it because we want to keep the mapping to the original data - this is deprecated scrapper. 

The other is scrapper that failing always (all cases) and so we are good with it to be marked as failing always - but we need each test to make sure it still in the gov site.

The source for what is in the website should always be the files stored in the repo.

When running tests the expectation from deprecated scrappers is not to be run, while  for the other scrapper is turon and accept failed. If it's not failing although expected, you should fail it too. 


so before the PR 
"QUIK",
        "VICTORY"
"CITY_MARKET_KIRYATGAT",
        "NETIV_HASED",
        
are marked fail and it ok.


after the changed should marked at deprected 

"QUIK",
        "VICTORY"
        
and 

"CITY_MARKET_KIRYATGAT",
        "NETIV_HASED",
        
should failed since th gov il site link as changes.


the this should tirgger a change to the scraper configuration.

**Comment by erlichsefi** [comment]
Reworked around three kinds, with cached `cpfta_prices_regulations` as the listing source:

| Kind | Scrapers | Tests | Production |
|---|---|---|---|
| **Edge-case** | SuperYuda, Cofix, Meshmat, … | Run; missing files OK only for `failire_valid()` | Run unless that edge case hits |
| **Always-failing** | `CITY_MARKET_KIRYATGAT`, `NETIV_HASED` | Run and **accept empty**; **fail if files appear** | Skipped (`failire_valid` always true) |
| **Deprecated** | `QUIK`, `VICTORY` | Not run (not in factory) | Not run |

`NETIV_HASED` scraper URL is now `https://app.netiv-hesed.com/` to match the cached listing. `always_failing_url_drift()` fails if an always-failing scraper's configured host is missing from that HTML — that is the signal to change scraper configuration (or mark it deprecated).

File: il_supermarket_scarper/scraper_stability.py [hunk 6]
```diff
@@ -1,20 +1,61 @@
...
+    scraper_cls = _scraper_class_for(name)
+    with tempfile.TemporaryDirectory() as tmp:
+        instance = scraper_cls(file_output=DiskFileOutput(storage_path=tmp))
+        url = getattr(instance, "url", None)
+        if url:
+            return url
+        ftp_host = getattr(instance, "ftp_host", None)
+        if ftp_host:
+            ftp_path = getattr(instance, "ftp_path", "") or ""
+            return f"ftp://{ftp_host}{ftp_path}"
```
Comment (erlichsefi) [inline il_supermarket_scarper/scraper_stability.py]: we should move this logic into the class.
